test/test_individual.py

```diff
@@ -1,139 +1,165 @@
+            "n_rows": 1,
+            "levels_back": 2,
+        },
+        primitives=(cgp.Add, cgp.Parameter),
+        dna=[
+            ID_INPUT_NODE,
+            ID_NON_CODING_GENE,
+            ID_NON_CODING_GENE,
+            1,
+            0,
+            0,
+            0,
+            0,
+            1,
+            ID_OUTPUT_NODE,
+            2,
+            ID_NON_CODING_GENE,
+        ],
+        target_function=lambda x, c: x + c,
+    )
+]
+FuncArgs = namedtuple("FuncArgs", ["x", "c"])
+
+func_args_list = [FuncArgs(x=[3.0, 5.0], c=1.0), FuncArgs(x=[3.0], c=2.0)]
+
+
+def _create_genome(genome_params, primitives, dna):
+    genome = cgp.Genome(**genome_params, primitives=primitives)
+    genome.dna = dna
+    return genome
+
+
+def _create_individual(fitness, genome, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return IndividualSingleGenome(fitness, genome)
+    elif individual_type == "MultiGenome":
+        return IndividualMultiGenome(fitness, [genome])
+    else:
+        raise NotImplementedError
+
+
+def _unpack_evaluation(value, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return value
+    elif individual_type == "MultiGenome":
+        return value[0]
+    else:
+        raise NotImplementedError
+
+
+def _unpack_genome(individual, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return individual.genome
+    elif individual_type == "MultiGenome":
+        return individual.genome[0]
+    else:
+        raise NotImplementedError
+
+
+@pytest.mark.parametrize("individual_type", ["SingleGenome", "MultiGenome"])
+def test_pickle_individual(individual_type):
 
     primitives = (cgp.Add,)
     genome = cgp.Genome(1, 1, 1, 1, 1, primitives)
-    individual = IndividualSingleGenome(None, genome)
+    individual = _create_individual(None, genome, individual_type)
 
     with open("individual.pkl", "wb") as f:
         pickle.dump(individual, f)
 
 
-def test_individual_with_parameter_python():
+@pytest.mark.parametrize("individual_type", ["SingleGenome", "MultiGenome"])
+@pytest.mark.parametrize("params", params_list)
+@pytest.mark.parametrize("func_args", func_args_list)
+def test_individual_with_parameter_python(individual_type, params, func_args):
+    genome_params, primitives, dna, target_function = params
+    genome = _create_genome(genome_params, primitives, dna)
+    individual = _create_individual(None, genome, individual_type)
 
-    primitives = (cgp.Add, cgp.Parameter)
-    genome = cgp.Genome(1, 1, 2, 1, 2, primitives)
-    # f(x) = x + c
-    genome.dna = [
-        ID_INPUT_NODE,
-        ID_NON_CODING_GENE,
-        ID_NON_CODING_GENE,
-        1,
-        0,
-        0,
-        0,
-        0,
-        1,
-        ID_OUTPUT_NODE,
-        2,
-        ID_NON_CODING_GENE,
-    ]
-    individual = IndividualSingleGenome(None, genome)
+    x, c = func_args.x, func_args.c
```

Comment:
i think an `assert` got lost that made sure than before setting the parameter in the next line, it's value is equal to the initial value, `1.0`. this also applies for more tests below.

Reply:
Sorry, I don't see that `assert` in the old code (left side of here). As far as I can see, the asserts only tested for the function evaluation, e.g. in line 47 and 55 of the old version.

Reply:
yes, you are right. i was misinterpreting the two consecutive asserts in the old cold: in both cases the parameter is assigned and then the output is checked. since you implemented this via `parametrized` i got a bit confused. since defaults are checked in `test_nodes`, never mind, everything's fine here :+1: